### ISSUE TITLE
Different references for the same PsiElement throws "Non-idempotent computation" exception

### ISSUE THREAD

**Issue by adelf** [ekvedaras/laravel-query-intellij#42]
```
java.lang.Throwable: Non-idempotent computation: it returns different results when invoked multiple times or on different threads:
  1 != 2
  which is length of [com.laravel_idea.plugin.eloquent.b.d@4cce9a9a] and [dev.ekvedaras.laravelquery.reference.ColumnPsiReference(StringLiteralExpressionImpl: 'vip':(1,4)), com.laravel_idea.plugin.eloquent.b.d@6a5ef7c8]

Recomputation gives com.intellij.util.CachedValueBase$Data@309762ce which is equivalent to 'fresh'
Recomputation log:
  dev.ekvedaras.laravelquery.reference.ColumnReferenceProvider@11ecc6a8 returned [dev.ekvedaras.laravelquery.reference.ColumnPsiReference(StringLiteralExpressionImpl: 'vip':(1,4))]
  com.laravel_idea.plugin.a.b@15486265 returned [com.laravel_idea.plugin.eloquent.b.d@2d6a201c]
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:161)
	at com.intellij.util.IdempotenceChecker.reportFailure(IdempotenceChecker.java:90)
	at com.intellij.util.IdempotenceChecker.checkEquivalence(IdempotenceChecker.java:77)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:237)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:72)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:155)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:121)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry.getReferencesFromProviders(ReferenceProvidersRegistry.java:38)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry.getReferencesFromProviders(ReferenceProvidersRegistry.java:31)
	at com.jetbrains.php.lang.psi.elements.impl.StringLiteralExpressionImpl.getReferences(StringLiteralExpressionImpl.java:432)
	at com.laravel_idea.plugin.extension.inspections.LaravelUnknownViewInspection$buildVisitor$1.visitPhpStringLiteralExpression(LaravelUnknownViewInspection.kt:32)
```
When my plugin asks for references for the StringLiteralExpression element this error appears. As you see, one time your plugin adds references for this element, another time - doesn't.

Actually, I have a lot of these issues: 
https://github.com/laravel-idea/plugin/issues/296
https://github.com/laravel-idea/plugin/issues/262

I wanted to make a PR here, but couldn't understand this code: `if (isResolving.contains(element)) {`. I'm sure this is the reason for these exceptions.

**Comment by ekvedaras** [comment]
Hey. That code was added to avoid infinite nesting in some specific cases. Not sure how to solve it at the moment 🤔 But I will take a look.

**Comment by ekvedaras** [comment]
Released a hotfix [v2.0.2](https://github.com/ekvedaras/laravel-query-intellij/releases/tag/v2.0.2). Not 100% if it will solve the exception, but I think so. If not, I can remove model create/update/etc methods from being inspected since that is one place where these plugins overlap.

**Comment by ekvedaras** [comment]
Should be resolved with [v2.0.3](https://github.com/ekvedaras/laravel-query-intellij/releases/tag/v2.0.3).

**Comment by adelf** [comment]
Thank you, Ernestas.
